cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -269,29 +285,176 @@ func copyFromGlobHelper(src, dst, globEx string) error {
+														|                              |
+				     							success |                              |
+														|                              |
+														v                              v
+	fail or        +-------------------+  fail,     +----+------+  fail,       +--------+-------+
+	not needed     |    locate save    |  unencrypt |unlock data|  encrypted   | unlock data w/ |
+	+--------------+    unencrypted    +<-----------+w/ run key +--------------+ fallback key   +-------+
+	|              |                   |            |           |              |                |       |
+	|              +--------+----------+            +-----+-----+              +--------+-------+       |
+	|                       |                             |                             |               |
+	|                       |success                      |success                      |               |
+	|                       |                             |                    success  |        fail   |
+	v                       v                             v                             |               |
++---+---+           +-------+----+                +-------+----+                        |               |
+|       |           | mount      |       success  | mount data |                        |               |
+| done  +<----------+ save       |      +---------+            +<-----------------------+               |
+|       |           |            |      |         |            |                                        |
++--+----+           +----+-------+      |         +----------+-+                                        |
+   ^                     ^              |                    |                                          |
+   |                     | success      v                    |                                          |
+   |                     |     +--------+----+   fail        |fail                                      |
+   |                     |     | unlock save +--------+      |                                          |
+   |                     +-----+ w/ run key  |        v      v                                          |
+   |                     ^     +-------------+   +----+------+-----+                                    |
+   |                     |                       | unlock save     |                                    |
+   |                     |                       | w/ fallback key +------------------------------------+
+   |                     +-----------------------+                 |
+   |                             success         +-------+---------+
+   |                                                     |
+   |                                                     |
+   |                                                     |
+   +-----------------------------------------------------+
+						fail
+
+*/
+type stateMachine struct {
```

Comment:
good point, we don't currently need the context and to test the state machine we can load it up with whatever state we need since we will be calling individual steps one at a time and can setup whatever state we need. I think this may simplify some of the other awkward TODO's

Reply:
done, merged the context type into the stateMachine